File: DarkOrbit 10.0/Net/netty/commands/WindowSettingsModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Ow.Utils;

namespace Ow.Net.netty.commands
{
    class WindowSettingsModule
    {
        public const short ID = 1110;

        public bool HideAllWindows = false;
        public int Scale = 0;
        public String BarState = "";

        public WindowSettingsModule(int param1, String param2, bool param3)
        {
            this.Scale = param1;
            this.BarState = param2;
            this.HideAllWindows = param3;
        }

        public byte[] write()
        {
            var param1 = new ByteArray(ID);
            param1.writeUTF(this.BarState);
            param1.writeInt(this.Scale << 13 | this.Scale >> 19);
            param1.writeBoolean(this.HideAllWindows);
            param1.writeShort(-27662);
            param1.writeShort(15301);
            return param1.Message.ToArray();
        }
    }
}

File: DarkOrbit 9.0/DarkOrbit 9.0 Emulator/src/simulator/netty/serverCommands/WindowSettingsModule.java
package simulator.netty.serverCommands;

import java.io.DataOutputStream;
import java.io.IOException;

import simulator.netty.ServerCommand;

public class WindowSettingsModule
        implements ServerCommand {

    public static int     ID              = 319;
    public        boolean mHideAllWindows = false;
    public        int     mScale          = 0;
    public        String  mBarState       = "";

    public WindowSettingsModule(int param1, String param2, boolean param3) {
        this.mScale = param1;
        this.mBarState = param2;
        this.mHideAllWindows = param3;
    }

    public void write(DataOutputStream param1) {
        try {
            param1.writeShort(ID);
            this.writeInternal(param1);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    protected void writeInternal(DataOutputStream param1) {
        try {
            param1.writeShort(-8579);
            param1.writeBoolean(this.mHideAllWindows);
            param1.writeUTF(this.mBarState);
            param1.writeInt(this.mScale << 9 | this.mScale >>> 23);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}